### PR TITLE
[native] Make memory reclaim timeout configurable

### PR DESCRIPTION
**Pull request by arhimondr**
## Description

Allow changing memory reclaim timeout

## Motivation and Context

Sometimes it is useful to increase it or set it to 0 to effectively disable it and have the process wait forever for further diagnosis

## Impact

N/A

## Test Plan

- Run TestPrestoSparkNativeAggregations
- Check the configuration property is added to config.properties and applied by the native process

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/common/Configs.cpp [hunk 1]
```diff
@@ -463,6 +463,12 @@ uint64_t SystemConfig::memoryPoolTransferCapacity() const {
       .value_or(kMemoryPoolTransferCapacityDefault);
 }
 
+uint64_t SystemConfig::memoryReclaimWaitMs() const {
+  static constexpr uint64_t kMemoryReclaimWaitMsDefault = {300'000};
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/common/Configs.cpp:467]:
```
nit: static constexpr uint64_t kMemoryReclaimWaitMsDefault ... // 5 mins.
```

**Review by xiaoxmeng — APPROVED** [review]
@arhimondr LGTM. Thanks!

**Review by xiaoxmeng — APPROVED** [review]

**Review by xiaoxmeng — APPROVED** [review]

File: presto-native-execution/presto_cpp/main/common/Configs.h [hunk 2]
```diff
@@ -316,6 +316,15 @@ class SystemConfig : public ConfigBase {
   /// NOTE: this config only applies if the memory arbitration has been enabled.
   static constexpr std::string_view kMemoryPoolTransferCapacity{
       "memory-pool-transfer-capacity"};
+
+  /// Specifies the max time to wait for memory reclaim by arbitration. The
+  /// memory reclaim might fail if the max wait time has exceeded. If it is
+  /// zero, then there is no timeout.
+  ///
+  /// NOTE: this config only applies if the memory arbitration has been enabled.
+  static constexpr std::string_view kMemoryReclaimWaitMs{
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/common/Configs.h:325]: Would you update documentation to add this new property? CC: @xiaoxmeng

**Review by mbasmanova — APPROVED** [review]
Looks good, but please update documentation.

File: presto-native-execution/presto_cpp/main/common/Configs.h (hunk 2 above)
Comment (arhimondr) [reply presto-native-execution/presto_cpp/main/common/Configs.h:325]: Let's follow up on this

**Pull request merged**